### DISCUSSION TITLE
Best way of performing algebraic operations on textures?

### DISCUSSION THREAD

**Discussion by georgy95** [mitsuba-renderer/mitsuba3#874] [Q&A] [answered]
Currently, most materials support Texture `OR` RGB albedo inputs. I want to change this behaviour to `AND` (`RGB * Texture`) i.e. tint this texture, for example, multiply the bitmap by `[2, 2, 2]` to make it brighter. 

As it stands, I could edit the texture itself and load a new texture. Or convert the texture to a tensor and apply the multiplication by a 3Vector. Both of these methods seem very slow and are computed on the CPU.

What is a better way of handling this:
- Create a custom material that separates `RGB` and `Texture` types? I.e. add `base_color_tint + base_color_texture` parameters.
- Create a new texture plugin, which accepts `scale` parameter
- Current way of scaling the Tensor

**Comment by njroussel** [comment, marked as answer]
Hi @georgy95 

The second option is the most generic one, I'd recommend that. (To be clear all 3 approaches will work just fine and will have similar performance). 
We will "soon" be working on adding some basic arithmetic plugins for textures. This will mostly be used for material conversions from Blender into Mitsuba. You can expect a similar set of shading nodes.

**Reply by georgy95** [reply]
Thank you. And awesome to hear!!
